R-package/tests/testthat/test_utils.R

```diff
@@ -68,3 +77,50 @@ test_that("lgb.last_error() correctly returns errors from the C++ side", {
         dvalid1$construct()
     }, regexp = "[LightGBM] [Fatal] Length of label is not same with #data", fixed = TRUE)
 })
+
+context("lgb.check.eval")
+
+test_that("lgb.check.eval works as expected with no metric", {
+    params <- lgb.check.eval(
+        params = list(device = "cpu")
+        , eval = "binary_error"
+    )
+    expect_named(params, c("device", "metric"))
+    expect_identical(params[["metric"]], list("binary_error"))
+})
+
+test_that("lgb.check.eval adds eval to metric in params", {
+    params <- lgb.check.eval(
+        params = list(metric = "auc")
+        , eval = "binary_error"
+    )
+    expect_named(params, "metric")
+    expect_identical(params[["metric"]], list("auc", "binary_error"))
+})
+
+test_that("lgb.check.eval adds eval to metric in params", {
+    params <- lgb.check.eval(
+        params = list(metric = "auc")
+        , eval = "binary_error"
+    )
+    expect_named(params, "metric")
+    expect_identical(params[["metric"]], list("auc", "binary_error"))
+})
```

Comment:
@jameslamb These two tests look identical. Is it expected or just copy-paste issue?

Reply:
oh yes! Must be a copy-paste mistake. This is where I miss Python tests, where each test is a named function and a linter can warn you if you have two with the same name